src/Compatibility/Core/src/iOS/Renderers/NavigationRenderer.cs

```diff
@@ -40,7 +40,7 @@ public class NavigationRenderer : UINavigationController, IVisualElementRenderer
 		[Preserve(Conditional = true)]
 		public NavigationRenderer() : base(typeof(FormsNavigationBar), null)
 		{
-			MessagingCenter.Subscribe<IVisualElementRenderer>(this, UpdateToolbarButtons, sender =>
+			WeakReferenceMessenger.Default.Register<UpdateToolBarButtonsMessage>(this, (receiver, message) => 
```

Comment:
This seems to be ignoring the recipient too and introducing a closure:

```suggestion
			WeakReferenceMessenger.Default.Register<NavigationRenderer, UpdateToolBarButtonsMessage>(this, (receiver, message) => 
```

Plus changes in the handler to invoke the instance methods on `receiver` directly.
Also, rename to `(r, m)` for consistency? Same in some other handlers above.